js/post/update_post.js

```diff
@@ -316,74 +312,97 @@ function fill400s() {
 
     
     // clear textboxes
-    for (k = 0; k < 3; k++) {
+    for (var k = 0; k < 3; k++) {
         spec400s[k].value = '';
+        spec400s[k].readOnly = false;
         if (k < 1) {
             maj400s[k].value = '';
+            maj400s[k].readOnly = false;
         }
         min400s[k].value = '';
+        min400s[k].readOnly = false;
     }
     
-
     // fill courses that have been selected
-    for (var m = 0; m < active400s.length; m++) {
-        if ((index400 == active400s.length) || (i === 3)) {
-            break;
-        }
-        spec400s[i].value = active400s[index400];
-        if (i < 1) {
-            maj400s[i].value = active400s[index400];
-        }
-        min400s[i].value = active400s[index400]; 
-        i += 1;
-        index400 += 1;
-    }
+    fill300and400Textboxes(active400s.length, 1, '400', '400', i, spec400s, maj400s, min400s);
+
 }
  
-
+/**
+ * Autofills the textboxes for the extra 300+ credits category
+**/
 function fillExtra() {
-    'use-strict'
+    'use strict';
 
     var i = 0;
     var spec_extra = $('#spec_extra')[0].getElementsByTagName('input');
     var maj_extra = $('#maj_extra')[0].getElementsByTagName('input');
 
     for (var k = 0; k < 4; k++) {
+
+        // clear text boxes
         if (spec_extra[k].value.indexOf('MAT') === -1 && spec_extra[k].value.indexOf('STA') === -1) {
             spec_extra[k].value = '';
+            spec_extra[k].readOnly = false;
         } if (k < 3) {
             if (maj_extra[k].value.indexOf('MAT') === -1 && maj_extra[k].value.indexOf('STA') === -1) {
                 maj_extra[k].value = '';
+                maj_extra[k].readOnly = false;
+            }
+        } 
+
+        // add credit count for MAT and STA courses
+        if (spec_extra[k].value.indexOf('MAT') > -1 || spec_extra[k].value.indexOf('STA') > -1) {
+            creditCount300and400['spec'] += 0.5;
+            filled_textboxes_extra['spec'] += 1;
+        } if (k < 3) {
```

Comment:
If you mean `if (k < 3)` (sometimes I can't tell which line number it's referring to), I actually think I need to keep this as an `if`. It's a bound check because a major has less 400 level courses than a specialist does. But I'll change the positioning to show so.